src/enum_net.cpp

```diff
@@ -158,13 +158,20 @@ namespace libtorrent { namespace
 
 		do
 		{
-			int read_len = recv(sock, buf, bufsize - msg_len, 0);
+			int read_len = int(recv(sock, buf, bufsize - msg_len, 0));
 			if (read_len < 0) return -1;
 
 			nl_hdr = reinterpret_cast<nlmsghdr*>(buf);
 
+#if (TORRENT_ANDROID && defined __clang__)
+#pragma clang diagnostic push
```

Comment:
would you mind adding a comment here why the sign-compare needs to be disabled? I realize I haven't commented in similar situations, but I think if this is becoming a more established pattern, for more platforms, it might be a good idea